python/python/lancedb/query.py

```diff
@@ -1644,7 +1634,7 @@ def nearest_to(
         if (
             isinstance(query_vector, list)
             and len(query_vector) > 0
-            and not isinstance(query_vector[0], (float, int))
+            and not np.isrealobj(query_vector[0])
```

Comment:
`isrealobj` seems like an odd choice.  We want to test if query_vector[0] is "array like" in some way right?

Reply:
You are right, changed to check it is a list / np.ndarray / pa.array